src/usdb_syncer/pdf.py

```diff
@@ -1,40 +1,217 @@
 """Generates a PDF from the passed song list."""
 
-# maybe reportlab is better suited?
-import datetime
+from collections import defaultdict
 from collections.abc import Iterable
-from typing import Any
+from typing import assert_never
 
-from pdfme import build_pdf
+from reportlab.lib import colors
+from reportlab.lib.pagesizes import A3, A4, A5, LEGAL, LETTER, landscape
+from reportlab.lib.styles import ParagraphStyle
+from reportlab.lib.units import mm
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+from reportlab.pdfgen.canvas import Canvas
+from reportlab.platypus import BaseDocTemplate, Flowable, Frame, PageTemplate, Paragraph
 
 from usdb_syncer import SongId
+from usdb_syncer.gui.song_table.column import Column
+from usdb_syncer.settings import ReportOrientation, ReportPagesize
 from usdb_syncer.usdb_song import UsdbSong
 
+pdfmetrics.registerFont(
+    TTFont("NotoSans-Black", "./src/usdb_syncer/gui/resources/NotoSans-Black.ttf")
```

Comment:
Paths should not be hardcoded strings but Path objects in AppPaths (at least the parent folder).